### PR TITLE
Gapped frame processing and calculates Estimated time of completion

### PR DESCRIPTION
**Pull request by AadityaDeshpande**
# New Changes in 3.0V
## Speed Improvement
**To improve speed, Idea is to process frames after a certain gap of frames between them**
This will improve the speed, as in version 2.0 it was processing every frame making it time-consuming.

feel free to contact me in case if you want to do some changes in the source code.😊😉
I'm happy to collaborate with you.😄

Regards,
Aaditya Deshpande

### CODE REVIEW

File: social distancing analyser 3.0.py [hunk 1]
```diff
@@ -0,0 +1,303 @@
...
+
+    #Processing Only Certain gap frames.
+
+    if counter!=1 and counter%gapFactor!=0:
+        frame = cv2.resize(frame, (720,480))
+        #cv2.imshow('Social distancing analyser', frame)
+        #cv2.waitKey(1)
+        #print("Non-Processed Frame ",counter)
+        #writer.write(frame)
+        layerOutputs = oldlayer
```
Comment (ShounakCy) [inline social distancing analyser 3.0.py:157]: the ide shows undefined name "oldlayer" here.

Comment (AadityaDeshpande) [reply social distancing analyser 3.0.py:157]: Thank you for your review, **just resolved this**, fixed some minor issues, (like file not found) and as the processing of each frame depends upon the processor so made some changes in the estimated time calculation. 😄

**Comment by Ank-Cha** [comment]
Hey! @AadityaDeshpande  Thank you for taking interest in my project.

Actually I have already developed V3.0 with integrated mask detection, and I have skipped frames in order to improve performance as the integration of mask detection severely affects the speed of processing.

I am a bit caught up in work which is the reason I haven't able to release V3.0 code (need to do a few more fixes in pixel mapping before committing).

I would really appreciate it if you can wait till I release v3.0 to suggest and contribute more features or optimizations to avoid complications.

**Comment by AadityaDeshpande** [comment]
Hi, @Ank-Cha , Thank you for reviewing my changes In code.
I would really like to see your V3.0 code soon.
Best of Luck 😄

Sorry for the late reply.

**Comment by ZIKO94ZIKO** [comment]
Hi @AadityaDeshpande thank you so much ,  could you please send me you update like a file.py in email.

**Comment by AadityaDeshpande** [comment]
Hi @ZIKO94ZIKO, the updated file is already there in my forked repo.

**Comment by ZIKO94ZIKO** [comment]
Ok sir , tell me sir how can be run this script  with GPU , because when i runnig the script  , it use just 7%with GPU performance ,Thanks.

**Comment by zongoalbert** [comment]
hello everyone , i have the some problem , i can't run it with GPU , your update @AadityaDeshpande  work well, but the soft need more acceleration , i  tested this line to work with tensorflow-gpu but the GPU always use just 10%  : 
`import tensorflow as tf
if tf.__version__ > '2':
    import tensorflow.compat.v1 as tf
    #tf.compat.v1.disable_v2_behavior()



def set_cpu_option():
    os.environ["CUDA_DEVICE_ORDER"] = "PCI_BUS_ID"  # see issue #152
    os.environ["CUDA_VISIBLE_DEVICES"] = ""
    os.environ["CUDA_VISIBLE_DEVICES"] = ""

def set_gpu_option(which_gpu):
    config = tf.compat.v1.ConfigProto()
    config.gpu_options.allow_growth = True
    config.log_device_placement = True
    config.gpu_options.visible_device_list= '0'
   
    sess = tf.Session(config=config) 

    tf.keras.backend.set_session(sess)

    return

set_gpu_option("0")
# or 
#set_cpu_option()`
Any help you could provide would be greatly appreciated , Many Thanks.
